lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -229,18 +226,42 @@ bool ProxyRenderDelegate::requiresUpdate(const MSubSceneContainer& container, co
     return true;
 }
 
+void ProxyRenderDelegate::_ClearRenderDelegate()
+{
+    delete _sceneDelegate;
+    _sceneDelegate = nullptr;
+    delete _taskController;
+    _taskController = nullptr;
+    delete _renderIndex;
+    _renderIndex = nullptr;
+    delete _renderDelegate;
+    _renderDelegate = nullptr;
+}
+
 //! \brief  One time initialization of this drawing routine
-void ProxyRenderDelegate::_InitRenderDelegate() {
-    // No need to run all the checks if we got till the end
-    if (_isInitialized())
-        return;
+void ProxyRenderDelegate::_InitRenderDelegate(MSubSceneContainer& container) {
 
     if (_proxyShape == nullptr)
         return;
 
-    if (!_usdStage) {
-        _usdStage = _proxyShape->getUsdStage();
+    UsdStageRefPtr newUsdStage = _proxyShape->getUsdStage();
+    if (_usdStage != newUsdStage)
+    {
+        _ClearRenderDelegate();
+
+        // delete everything so we stop drawing the old stage and draw the new one
+        _ClearRenderDelegate();
+        _dummyTasks.clear();
+
+        _isPopulated = false;
+        container.clear();
```

Comment:
You are saying this is not enough to clear the subscene state?

Reply:
According to the API tech docs tt is supposed to be enough to clear the subscene but it doesn't work correctly. Sometimes things get leftover and I have to ```ogs -reset``` to get rid of them all.

Reply:
Maybe that reveals some bugs in VP2?

Reply:
Just saw the other comment of yours above. I believe the correctness issue I am looking at are not a duplicate of this one. See my comments in JIRA for details.

Reply:
Yes I think there is something to fix in VP2. Maybe related to what you are already looking at @HdC-adsk. I'll put notes on which scenes I used to reproduce the issue in the JIRA story so we can validate it after you have a fix.

Reply:
I think we commented at the same time? The issue I am looking at is caused by transparent instances, so I don't think it is the same issue as leftover ghost objects.

Reply:
I agree I think some items are getting lost in consolidated world. I'll log an internal issue to investigate. It could be caused by our recent changes.